wip-asdpc-increase-collateral-utxo-age-requirement.md

```diff
@@ -0,0 +1,190 @@
+<pre>
+  WIP: WIP-ASDPC-INCREASE-COLLATERAL-UTXO-AGE-REQUIREMENT
+  Layer: Consensus (hard fork)
+  Title: Increase the age requirement for using transaction outputs as collateral 
+  Authors: Adán SDPC <adan@witnet.foundation>
+  Discussions-To: `#dev-lounge` channel on Witnet Community's Discord server
+  Status: Draft
+  Type: Standards Track
+  Created: 2022-10-26
+  License: BSD-2-Clause
+</pre>
+
+
+## Abstract
+
+This proposes an increase to the age requirement for using transaction outputs as collateral. The new required age
+will be 13440 protocol epochs, i.e. 1 week. This improvement aims at strengthening the protocol's security and
+economics.
+
+## Motivation and rationale
+
+### Current status
+
+_Collateralization_ in Witnet means that nodes that become eligible to take part in resolving a oracle request need to
+lock a certain amount of Wit coins upon publishing their commitment transaction. If their committed values pass the
+filters in the eventual _tally_ of the request, they get the collateralized coins back. If they do not pass the filters
+(aka _liars_), they lose the collateralized coins.
+
+To use a transaction output (_UTXO_) as collateral, it must have existed for more than 2,000 epochs (25 hours). We often
+call this the _UTXO age requirement_.
+
+Its current value was set in the Witnet testnet times (2019) and was never adjusted to the mainnet reality.
+
+Collateralized coins are refunded to their owners in the _tally_ transaction, with a new UTXO age of 0 epochs.
```

Comment:
I wasn't sure about this reset, but it looks correct. I believe I got confused because there is an exception for the UTXO age reset that applies to the change of the commit transaction. So if a node has a single UTXO with 1000 wits and they participate in a data request with 1 wit of collateral, the change output of 999 wits will have the same age as the original 1000 wit UTXO. But it is true that after the 1 wit is refunded, its age is reset.

That logic is implemented in the `update_utxo_diff` function of which I couldn't find any tests, but I believe it maintains the old UTXO age if all the inputs have the same PKH. Tally transactions have no inputs, so that exception does not apply.